run-build-functions.sh

```diff
@@ -98,23 +98,26 @@ run_yarn() {
   fi
 
 
-  echo "Installing NPM modules using Yarn version $(yarn --version)"
-  run_npm_set_temp
+  if install_deps package.json $NODE_VERSION $NETLIFY_CACHE_DIR/package-sha
```

Comment:
This is the same line of code that we currently use for `npm install`.

Reply:
It's interesting that we're running this against `package.json` instead of a lockfile if one is present.

I think mirroring the behavior of `run_npm` is the right idea for now, but let's keep an eye on it. Could imagine an edge case where someone changed `yarn.lock` but not `package.json`.

Reply:
Yes, I noticed this too.
Same here: I was thinking mirroring would be best for the moment, but in the future we might consider using `package-lock.json` or `yarn.lock` (if they exist, since those are optional).